contracts/CctpBridge.sol

```diff
@@ -0,0 +1,194 @@
+        uint relayerFee,
+        uint receivedRelayerFeeTokenAmount,
+        uint adminFeeTokenAmount
+    );
+
+    constructor(
+        uint chainId_,
+        uint chainPrecision_,
+        address tokenAddress,
+        address cctpMessenger_,
+        address cctpTransmitter_,
+        IGasOracle gasOracle_
+    ) GasUsage(gasOracle_) {
+        chainId = chainId_;
+        token = IERC20Metadata(tokenAddress);
+        uint tokenDecimals = token.decimals();
+        cctpMessenger = ITokenMessenger(cctpMessenger_);
+        cctpTransmitter = IMessageTransmitter(cctpTransmitter_);
+        token.approve(cctpMessenger_, type(uint256).max);
+        stableTokensForGasScalingFactor = 10 ** (ORACLE_PRECISION - tokenDecimals + chainPrecision_);
+        fromGasOracleScalingFactor = 10 ** (ORACLE_PRECISION - tokenDecimals);
+    }
+
+    function bridge(
+        uint amount,
+        bytes32 recipient,
+        uint destinationChainId,
+        uint relayerFeeTokenAmount
+    ) external payable {
+        require(amount > relayerFeeTokenAmount, "CctpBridge: Amount must be > relayerFeeTokenAmount");
+        require(recipient != 0, "CctpBridge: Recipient must be nonzero");
+        token.safeTransferFrom(msg.sender, address(this), amount);
+        uint gasFromStables = _getStableTokensValueInGas(relayerFeeTokenAmount);
+        uint relayerFee = this.getTransactionCost(destinationChainId);
+        require(msg.value + gasFromStables >= relayerFee, "CctpBridge: Not enough fee");
+        uint amountToSend = amount - relayerFeeTokenAmount;
+        uint adminFee = (amountToSend * adminFeeShareBP) / BP;
+        amountToSend -= adminFee;
+        uint32 destinationDomain = getDomainByChainId(destinationChainId);
+        uint64 nonce = cctpMessenger.depositForBurn(amountToSend, destinationDomain, recipient, address(token));
+        emit TokensSent(
+            amountToSend,
+            msg.sender,
+            recipient,
+            destinationChainId,
+            nonce,
+            msg.value,
+            gasFromStables,
+            relayerFee,
+            relayerFeeTokenAmount,
+            adminFee
+        );
+    }
+
+    function receiveTokens(address recipient, bytes calldata message, bytes calldata signature) external payable {
+        require(cctpTransmitter.receiveMessage(message, signature), "CctpBridge: Receive message failed");
+        // pass extra gas to the recipient
+        if (msg.value > 0) {
+            // ignore if passing extra gas failed
+            // solc-ignore-next-line unused-call-retval
+            payable(recipient).call{value: msg.value}("");
+        }
+    }
+
+    /**
+     * @notice Allows the admin to add new supported chain destination.
+     * @param chainId_ The chain ID of the destination to register.
+     * @param domain The domain of the destination to register.
+     */
+    function registerBridgeDestination(uint chainId_, uint32 domain) external onlyOwner {
```

Comment:
I recommend to add unregisterBridgeDestination method

Reply:
Fixed